### ISSUE TITLE
API documentation

### ISSUE THREAD

**Issue by weepy** [weepy/o_O#47]
really needs this. what's the best structure/tech

**Comment by Ricket** [comment]
I personally like [Docco](http://jashkenas.github.com/docco/). Backbone.js uses this for its [annotated source](http://documentcloud.github.com/backbone/docs/backbone.html). There is also [JSDoc](https://code.google.com/p/jsdoc-toolkit/), though I haven't used it.

**Comment by weepy** [comment]
Yes I think that's a good idea. I suppose we need annotated source _and_ API / guide docs.
